File: BAEKJOON/Python/10000/10800/10810.py
"""
Title : 공 넣기
Link : https://www.acmicpc.net/problem/10810
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    N, M = map(int, input().split())
    basket = [0] * N
    for _ in range(M):
        i, j, k = map(int, input().split())
        for idx in range(i - 1, j):
            basket[idx] = k
    print(*basket)

File: BAEKJOON/Javascript/10000/10800/10810.js
/*
Title : 공 넣기
Link : https://www.acmicpc.net/problem/10810
*/

const fs = require("fs");
const filepath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt";
const input = fs.readFileSync(filepath).toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const basket = new Array(n).fill(0);
for (let x = 0; x < m; x++) {
  const [i, j, k] = input[x + 1].split(" ").map(Number);
  for (let y = i; y <= j; y++) {
    basket[y - 1] = k;
  }
}
console.log(basket.join(" "));
